File: JY/8.28/타겟넘버.py
def solution(numbers, target):
    ans = 0
    n = len(numbers)
    stack = []
    
    stack.append([1])
    stack.append([2])
    
    while stack:
        node = stack.pop()
        if len(node) == n:
            s = 0
            for i in range(n):
                if node[i]==1:
                    s += numbers[i]
                if node[i]==2:
                    s -= numbers[i]
            if s == target:
                ans += 1
            continue
        stack.append(node+[1])
        stack.append(node+[2])
    return ans

File: HB/20200828/타겟넘버.cpp
#include <string>
#include <vector>

using namespace std;

int degree_of_total_method = 0;

void dfs(vector<int> &numbers, int &target, int sum, int n)
{
    if (n == numbers.size())
    {
        if (sum == target)
            degree_of_total_method++;
        return;
    }

    dfs(numbers, target, sum + numbers[n], n + 1);
    dfs(numbers, target, sum - numbers[n], n + 1);
}

int solution(vector<int> numbers, int target)
{
    int answer = 0;

    dfs(numbers, target, numbers[0], 1);
    dfs(numbers, target, -numbers[0], 1);

    answer = degree_of_total_method;

    return answer;
}